radix_sort.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+
+
+def int_radix(mylist):
+    '''
+    Sorts a list of ints by looking at the 1's place,
+    the 10's, 100's, etc.
+    '''
+    if len(mylist) <= 1:
+        return mylist
+    for i in mylist:
+        if (type(i) != int) or (i < 0):
+            raise TypeError('Please input valid integers')
+    if len(mylist) > 0:
+        max_list = max(mylist)
+        tens = 1
+        num_buckets = []
+        for i in range(10):
+            num_buckets.append([])
+        while max_list >= tens:
+            for item in mylist:
+                bucket = (item % (tens * 10) / tens)
+                num_buckets[bucket].append(item)
+            buff = []
+            for bucket in num_buckets:
+                while len(bucket) > 0:
+                    buff.append(bucket.pop(0))
```

Comment:
This is nasty slow.  `list.pop(0)` is an O(n) operation.  There's a much faster, more efficient way to extend a list with the items in another list.  Figure that out and use it instead.